### PR TITLE
RPN-1433: add batch request for items

### PR DESCRIPTION
**Pull request by mystdeim**

### CODE REVIEW

File: nft/api/src/test/kotlin/com/rarible/protocol/nft/api/e2e/items/ItemControllerFt.kt [hunk 1]
```diff
@@ -413,6 +414,21 @@ class ItemControllerFt : SpringContainerBaseTest() {
         assertEquals(NftItemRoyaltyListDto(listOf(NftItemRoyaltyDto(royalty._1, royalty._2.intValueExact()))), dto)
     }
 
+    @Test
+    fun `should get item by ids`() = runBlocking<Unit> {
+        val item = createItem()
+        itemRepository.save(item).awaitFirst()
+
+        val items = nftItemApiClient.getNftItemsByIds(NftItemIdsDto(listOf(item.id.decimalStringValue))).collectList().awaitFirst()
+        assertThat(items).hasSize(1)
```
Comment (atapin) [inline nft/api/src/test/kotlin/com/rarible/protocol/nft/api/e2e/items/ItemControllerFt.kt:423]: IMO it's better to separate assertions of service logic and conversion logic. Conversions can be tested in a separate unit tests

File: pom.xml [hunk 2]
```diff
@@ -34,7 +34,7 @@
         <rarible.core.version>2.2.2</rarible.core.version>
```
Comment (atapin) [inline pom.xml:34]: v2.2.4 will add `mapAsync`, `flatMapAsync` on iterables and maps

File: nft/api/src/main/kotlin/com/rarible/protocol/nft/api/service/item/ItemService.kt [hunk 3]
```diff
@@ -109,4 +105,18 @@ class ItemService(
             }
         }.awaitAll().sortedBy { it.item.date }.reversed()
     }
+
+    suspend fun search(list: Set<ItemId>): List<ExtendedItem> = coroutineScope {
+        val items = itemRepository.searchByIds(list)
+        toExtend(items)
+    }
+
+    suspend fun toExtend(items: List<Item>): List<ExtendedItem> = coroutineScope {
```
[the remark this answers is not in the document]
Comment (mystdeim) [reply nft/api/src/main/kotlin/com/rarible/protocol/nft/api/service/item/ItemService.kt:111]: fixed

File: pom.xml (hunk 2 above)
Comment (mystdeim) [reply pom.xml]: fixed

File: nft/api/src/test/kotlin/com/rarible/protocol/nft/api/e2e/items/ItemControllerFt.kt (hunk 1 above)
Comment (mystdeim) [reply nft/api/src/test/kotlin/com/rarible/protocol/nft/api/e2e/items/ItemControllerFt.kt:423]: Could you be more specific about how to rewrite this test?

Comment (atapin) [reply nft/api/src/test/kotlin/com/rarible/protocol/nft/api/e2e/items/ItemControllerFt.kt:423]: this is just my POV. I would leave `assertThat(items).hasSize(1)` as the last expression. All the rest lines would go to `ItemToDtoConverterTest`

Comment (yakischik) [reply nft/api/src/test/kotlin/com/rarible/protocol/nft/api/e2e/items/ItemControllerFt.kt:423]: I think, Andrey means there is no sense to assert secondary fields - to check id is enough. Other fields - responseibility of converter (and ideally there sohuld be test for converter to check other fields of DTO)

**Pull request merged**